core/net/rpl/rpl-ext-header.c

```diff
@@ -370,6 +374,12 @@ insert_srh_header()
   /* For simplicity, we use cmpri = cmpre */
   cmpri = 15;
   cmpre = 15;
+
+  if(node != NULL && node == root_node) {
```

Comment:
Ah, OK. Since I don't use an arrow (`->`) operation with `node` in that `if` block, we don't need a NULL-check for `node`. I agree that `node != NULL` is unnecessary.